tests/test_primary.py

```diff
@@ -266,85 +335,221 @@ def test_10_register_ecu_manifest(self):
     self.assertEqual([], TestPrimary.instance.nonces_to_send)
     self.assertEqual([], TestPrimary.instance.nonces_sent)
 
-    sample_ecu_manifest = {
-        "signatures": [{
-          "method": "ed25519",
-          "sig": "df043006d4322a386cf85a6761a96bb8c92b2a41f4a4201badb8aae6f6dc17ef930addfa96a3d17f20533a01c158a7a33e406dd8291382a1bbab772bd2fa9804",
-          "keyid": "49309f114b857e4b29bfbff1c1c75df59f154fbc45539b2eb30c8a867843b2cb"}],
-        "signed": {
-          "timeserver_time": "2016-10-14T16:06:03Z",
-          "installed_image": {
-            "filepath": "/file2.txt", "fileinfo": {
-              "hashes": {"sha256": "3910b632b105b1e03baa9780fc719db106f2040ebfe473c66710c7addbb2605a", "sha512": "e2ebe151d7f357fcc6b0789d9e029bbf13310e98bc4d15585c1e90ea37c2c7181306f834342080ef007d71439bdd03fb728186e6d1e9eb51fdddf16f76301cef"},
-            "length": 21}},
-          "previous_timeserver_time": "2016-10-14T16:06:03Z",
-          "ecu_serial": "ecu11111",
-          "attacks_detected": ""}}
+
+    # Load the manifests we'll use in these tests.
+    # Note that the .json and .der manifest samples aren't identical; they're
+    # signed over different data, so to get the JSON version of the DER
+    # manifests, we'll convert them.
+    # We'll always need the JSON encodings for testing, and we'll load the
+    # ASN.1/DER manifests only if we're in DER mode.
+    # 1: Correctly signed ECU manifest from ECU TCUdemocar
+    # 2: Correctly signed ECU manifest from ECU unknown_ecu
+    # 3: ECU Manifest from ECU TCUdemocar signed by the wrong key
+    #    (demo's Image Repo timestamp key in particular, instead of demo's
+    #     Secondary key)
+    # 4: Correctly signed ECU manifest from TCUdemocar w/ attack report
+
+    if tuf.conf.METADATA_FORMAT == 'json':
+      manifest1 = manifest1_json = json.load(open(os.path.join(TEST_DATA_DIR,
+          'flawed_manifests', 'em1_typical_ecu_manifest.json')))
+
+      manifest2 = manifest2_json = json.load(open(os.path.join(TEST_DATA_DIR,
+          'flawed_manifests', 'em2_unknown_ecu_manifest.json')))
+
+      manifest3 = manifest3_json = json.load(open(os.path.join(TEST_DATA_DIR,
+          'flawed_manifests', 'em3_ecu_manifest_signed_with_wrong_key.json')))
+
+      manifest4 = manifest4_json = json.load(open(os.path.join(TEST_DATA_DIR,
+          'flawed_manifests', 'em4_attack_detected_in_ecu_manifest.json')))
+
+    else:
+      assert tuf.conf.METADATA_FORMAT == 'der', 'Test code is flawed.'
```

Comment:
Consider using unittest's assert functions, to be consistent.

Reply:
My possibly idiosyncratic policy has been to use normal assertions in test code when I expect the cause of the error is due to test code being wrong rather than the code it tests being wrong. In this case, for example, the only reason I'd only expect METADATA_FORMAT to not be 'der' and not be 'json' is if a new metadata format has been added to the reference implementation and the test hasn't been updated yet. (The failure says something about the test code more than about the code tested.) I don't know if that's a common practice -- it just made sense to me as a policy -- so I'm very open to a suggestion that that doesn't make sense or is confusing. LMK (: